src/Util.h

```diff
@@ -472,8 +483,8 @@ std::unique_ptr<A> clone(const std::unique_ptr<A>& node) {
 }
 
 template <typename A>
-std::vector<std::unique_ptr<A>> clone(const std::vector<std::unique_ptr<A>>& xs) {
-    std::vector<std::unique_ptr<A>> ys;
+auto clone(const std::vector<A>& xs) {
```

Comment:
Can this be generalised to handle more than vectors? Vectors are easily our most common case so it's not an important change.

Reply:
Yep, just add a new templated overload.